Was written tests on mocha and jest for classes from homework 9

salopower opened the pull request.

salopower committed:
Was written tests on mocha and jest for classes from homework 9

salopower committed:
Fix - added all needed files

nzamishchak commented on home_works/home_work_10/mocha/tests/mocha-checkbox.test.spec.ts:
```diff
@@ -0,0 +1,30 @@
+        expect(checkbox.isDisabled).to.be.true;
+        checkbox.enable();
+        expect(checkbox.isDisabled).to.be.false;
+    });
```
в цьому тесті тестується одразу 2 методи. Юніт тести- це тести на мінімальну одиницю коду. Тому тут мало би бути 2 окремі тести.

nzamishchak commented on home_works/home_work_10/mocha/tests/mocha-checkbox.test.spec.ts:
```diff
@@ -0,0 +1,30 @@
+        checkbox.click();
+        checkbox.disable();
+        checkbox.click();
+    });
```
тут нічого не перевіряється
що саме тут малось на увазі?

nzamishchak reviewed (changes requested):
1. Тести одинакові для jest  і для mocha. З використанням одного з фреймворків допишіть тести на функції testUIElement і disableElement із вашої 9 дз, передаючи на вхід екземпляри різних класів. (щоб була можливість відтестувати функції їх треба трохи модифікувати, можете скопіювати в дз 10 index.ts і замість виведення в консоль зробити ретурн тексту)
2. Розділіть тести, що поєднують тестування двох методів(enable-disable)

salopower committed:
Correcting errors regarding comments

salopower committed:
Fixing errors with imports

nzamishchak commented on home_works/home_work_10/jest/tests/jest-test-ui-element.spec.ts:
```diff
@@ -0,0 +1,45 @@
+    it('Should test a Button element', () => {
+        const button = new Button('Submit');
+        const result = testUIElement(button);
+        expect(result).to.equal('The \'Submit\' element was clicked.');
```
сумніваюсь, що у Вас ці тести працюють. оскільки testUIElement 
```typescript
export function testUIElement(element: UIElement): void {
    element.click();
    console.log(element.getValue());
}
```

нічого не повертає. Щоб не руйнувати 9 роботу, я пропонувала Вам скопіювати цю  і іншу функцію в окремий файл цієї 10 домашньої і переробити функції, щоб вони повертали якесь значення. Це може бути щось ось таке
```typescript
import { UIElement } from '../../../home_work_9/src/ui-element';
import { Disableable } from '../../../home_work_9/src/disableable';

export function testUIElement(element: UIElement): string {
    element.click();
    return element.getValue();
}

export function disableElement(element: Disableable): string {
    element.disable();
    return `Element is disabled: ${String(element.isDisabled)}`;
}
```
І тести вже відповідно відкорегувати.

nzamishchak reviewed (changes requested):
треба поміняти функції, щоб можна було коректно їх відтестувати.

salopower committed:
refactor: update UIElement classes and tests to return values for better testability

- Updated `Button`, `Checkbox`, and `TextBox` classes to return strings instead of logging to console.
- Modified `click`, `disable`, `enable`, `setValue`, and `getValue` methods to return meaningful results.
- Updated Jest tests to check returned values instead of relying on console output.
- Adjusted `testUIElement` and `disableElement` functions to return testable results.
- Improved overall test coverage and maintainability.

nzamishchak reviewed (approved):
трохи намудровано, але вже краще).
бачу еслінт свариться на помилки. При можливості виправіть перед мерджанням.

salopower committed:
Merge branch 'master' into home_work_10
